File: training-center-client/src/app/Model/Student.ts
import { Enrollment } from "./Enrollment";


export interface Student {

  id: number;
  name : string;
  Email : string;
  age : number;
  major : string;
  phone : string;
  enrollment : Enrollment[];

}

export interface CreateStudent {


  name : string;
  Email : string;
  age : number;
  phone : string;
  major : string;
}

export interface UpdateStudent {


  name : string;
  Email : string;
  age : number;
  phone : string;
  major : string;
}

export interface StudentResponse{

  Id: number;
  Name : string;
  Email : string;
  Age : number;
  Phone : string;
  Major : string;

}

File: TrainingCenter.Api/Model/Student.cs
using System.ComponentModel.DataAnnotations;
using Microsoft.AspNetCore.Antiforgery;
namespace TRAININGCENTER.API.Model;

public class Student
{
  
    public int id {get; set;}
     
    public string name  {get; set;}=string.Empty;
     
     public string Email {get; set;} =string.Empty;

    
    public int age{get; set;} 
   
   public string major {get; set;}= string.Empty;
   
    public string phone {get;  set;} = string.Empty;

    public List<Enrollment> enrollment {get;set;} = new();


}